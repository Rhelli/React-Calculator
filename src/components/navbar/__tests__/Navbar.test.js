import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Navbar from '../Navbar.js';

describe('The Navbar Appearance', () => {
  it('Matches the Navbar Snapshot', () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Contains the Navbar title container and text', () => {
    const { container, getByText } = render(<Navbar />);
    expect(container.firstChild.firstChild.classList.contains('navbarTitle')).toBe(true);
    getByText('Math Magicians');
  });

  it('Contains the navbar links container and links', () => {
    const { container, getByText } = render(<Navbar />);
    expect(container.firstChild.lastChild.classList.contains('navbarLinks')).toBe(true);
    getByText('Home');
    getByText('Calculator');
    getByText('Quote');
  });
});
