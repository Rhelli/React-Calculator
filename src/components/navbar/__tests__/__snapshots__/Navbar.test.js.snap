// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Navbar Appearance Matches the Navbar Snapshot 1`] = `
<div
  className="navbarContainer"
>
  <div
    className="navbarTitle"
  >
    <h1>
      Math Magicians
    </h1>
  </div>
  <div
    className="navbarLinks"
  >
    <a
      href="/"
    >
      Home 
    </a>
    <span>
       | 
    </span>
    <a
      href="/Calculator"
    >
      Calculator 
    </a>
    <span>
       | 
    </span>
    <a
      href="/Quote"
    >
      Quote 
    </a>
  </div>
</div>
`;
